src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,175 @@
+        ///// <param name="value">The value</param>
+        ///// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        //public ComponentParameterBuilder<TComponent> AddCascading<TValue>(Expression<Func<TComponent, TValue>> parameterSelector, [DisallowNull] TValue value)
+        //{
+        //    if (parameterSelector == null)
+        //    {
+        //        throw new ArgumentNullException(nameof(parameterSelector));
+        //    }
+
+        //    if (value == null)
+        //    {
+        //        throw new ArgumentNullException(nameof(value));
+        //    }
+
+        //    AddParameterToList(ComponentParameter.CreateCascadingValue(GetParameterNameFromExpression(parameterSelector), value));
+        //    return this;
+        //}
+
+        /// <summary>
+        /// Add an unnamed cascading parameter with a value to this builder.
+        /// </summary>
+        /// <param name="value">The value</param>
+        /// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        public ComponentParameterBuilder<TComponent> Add(object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            AddParameterToList(ComponentParameter.CreateCascadingValue(null, value));
+            return this;
+        }
+
+        /// <summary>
+        /// Add a non generic <see cref="EventCallback"/> parameter with a value for this builder.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector</param>
+        /// <param name="callback">The event callback.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback>> parameterSelector, Func<Task> callback)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EventCallback.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Add a generic <see cref="EventCallback"/> parameter with a value for this builder.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector</param>
+        /// <param name="callback">The event callback.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback<EventArgs>>> parameterSelector, Func<EventArgs, Task> callback)

```

Comment:
```suggestion
        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback<T>>> parameterSelector, Func<T, Task> callback)
```
EventCallback can take any value, doesnt have to be a EventArgs type, as far as I know.